### PR TITLE
Add session-related endpoints

### PR DESCRIPTION
**Pull request by cmatheson**
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: spec/support/fixtures/vcr_cassettes/user_management/authenticate_with_code/valid.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ http_interactions:
           - cloudflare
       body:
         encoding: ASCII-8BIT
-        string: '{"user":{"object":"user","id":"user_01H93ZY4F80YZRRS6N59Z2HFVS","email":"test@workos.app","email_verified":false,"first_name":"Lucille","last_name":"Bluth","created_at":"2023-08-30T19:50:13.214Z","updated_at":"2023-08-30T19:50:13.214Z","user_type":"managed","sso_profile_id":"prof_01H93ZTVWYPAT4RKDSPFPPXH0J"}}'
+        string: '{"user":{"object":"user","id":"user_01H93ZY4F80YZRRS6N59Z2HFVS","email":"test@workos.app","email_verified":false,"first_name":"Lucille","last_name":"Bluth","created_at":"2023-08-30T19:50:13.214Z","updated_at":"2023-08-30T19:50:13.214Z","user_type":"managed","sso_profile_id":"prof_01H93ZTVWYPAT4RKDSPFPPXH0J"},"access_token":"access_token","refresh_token":"refresh_token"}'
```
Comment (cmatheson) [inline spec/support/fixtures/vcr_cassettes/user_management/authenticate_with_code/valid.yml:79]: i kind of hate vcr and i manually edited/added all of these responses. this is not technically the right way to do this.

File: lib/workos/refresh_authentication_response.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+# typed: true
+
+module WorkOS
+  # The AuthenticationResponse class represents an Authentication Response as well as an corresponding
+  # response data that can later be appended on.
```
Comment (mthadley) [inline lib/workos/refresh_authentication_response.rb:6]: What does "an corresponding response data that can later be appended on" refer to?

File: lib/workos/user_management.rb [hunk 3]
```diff
@@ -554,6 +594,66 @@ def authenticate_with_email_verification(
...
+      end
+
+      # Revokes a session
+      #
+      # @param [String] session_id The session ID can be found in the `sid`
+      #   claim of the access token
+      sig do
+        params(
+          session_id: String,
+        ).returns(NilClass)
```
Comment (mthadley) [inline lib/workos/user_management.rb:625]: Should this use [`.void`](https://sorbet.org/docs/sigs#returns--void-annotating-return-types) instead?

File: lib/workos/refresh_authentication_response.rb (hunk 2 above)
Comment (cmatheson) [reply lib/workos/refresh_authentication_response.rb:6]: ohhhh... this is a complete copy/paste from AuthenticationResponse. i'll come up with better wording :grimacing:

File: lib/workos/user_management.rb (hunk 3 above)
Comment (cmatheson) [reply lib/workos/user_management.rb:625]: ahhh, i knew there must be something like this. will fix.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -26,6 +26,12 @@
 VCR.configure do |config|
   config.cassette_library_dir = 'spec/support/fixtures/vcr_cassettes'
   config.filter_sensitive_data('<API_KEY>') { WorkOS.config.key }
+  config.filter_sensitive_data('<ACCESS_TOKEN>', :token) do |interaction|
+    JSON.parse(interaction.response.body)['access_token']
+  end
+  config.filter_sensitive_data('<REFRESH_TOKEN>', :token) do |interaction|
+    JSON.parse(interaction.response.body)['refresh_token']
+  end
```
Comment (mthadley) [inline spec/spec_helper.rb:34]: It'd be nice if this was the default and test authors didn't need to remember to opt-in, though I don't know the best way to implement that behavior without it also being inconsistent.

Comment (cmatheson) [reply spec/spec_helper.rb:34]: ugh, i agree. i tried just having the handler work without tags but it errors out if you don't return something. (i guess i could return some bogus value?). this seemed to be the "vcr" way to handle this situation.

**Pull request merged**